src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -72,13 +72,66 @@ def _determine_rgw_addr():
         raise LookupError('No RGW daemon found.')
 
     addr = daemon['addr'].split(':')[0]
-    match = re.search(r'port=(\d+)', daemon['metadata']['frontend_config#0'])
-    if match:
-        port = int(match.group(1))
-    else:
+    port, ssl = _parse_civetweb_config(daemon['metadata']['frontend_config#0'])
+    if port is None:
         raise LookupError('Failed to determine RGW port')
 
-    return addr, port
+    return addr, port, ssl
+
+
+def _parse_civetweb_config(config):
+    """
+    Get the port the RGW is running on. Due the complexity of the
+    syntax not all variations are supported.
+
+    Get more details about the configuration syntax here:
+    https://civetweb.github.io/civetweb/UserManual.html
+
+    >>> _parse_civetweb_config('civetweb port=8000')
+    (8000, False)
+
+    >>> _parse_civetweb_config('civetweb port=8000s')
+    (8000, True)
+
+    >>> _parse_civetweb_config('civetweb port=192.0.2.3:80')
+    (80, False)
+
+    >>> _parse_civetweb_config('civetweb port=172.5.2.51:8080s')
+    (8080, True)
+
+    >>> _parse_civetweb_config('civetweb port=[::]:8080')
+    (8080, False)
+
+    >>> _parse_civetweb_config('civetweb port=ip6-localhost:80s')
+    (80, True)
+
+    >>> _parse_civetweb_config('civetweb port=[2001:0db8::1234]:80')
+    (80, False)
+
+    >>> _parse_civetweb_config('civetweb port=[::1]:8443s')
+    (8443, True)
+
+    >>> _parse_civetweb_config('civetweb port=xyz')
+    (None, False)
+
+    >>> _parse_civetweb_config('civetweb')
+    (None, False)
+
+    :param config: The configuration string to parse.
+    :type config: str
+    :return: A tuple containing the port number and the information
+             whether SSL is used.
+    :rtype: (int, boolean)
+    """
+    port = None
+    ssl = False
+    match = re.search(r'port=(.*:)?(\d+)(s)?', config)
+    if match:
+        port = int(match.group(2))
+        ssl = match.group(3) == 's'
+    else:
+        logger.error('Failed to parse RGW port from \'%s\'.', config)
```

Comment:
Would it make sense to move the `LookupError` from above into this function instead of just logging the error here?

Reply:
I implemented it this way because this simplifies the doctests and i wanted to do the exception handling centralized in one function.

Reply:
You can also use Doctests with exceptons.

Reply:
Done